sources/api/schnauzer/src/helpers/mod.rs

```diff
@@ -535,6 +542,77 @@ pub fn tuf_prefix(
     Ok(())
 }
 
+/// The `aws_config` helper is used to create an AWS config file
+/// with `use_fips_endpoint` value set based on if the variant is FIPS enabled.
+///
+/// # Fallback
+///
+/// If this helper runs and `settings.aws.config` is set,
+/// the helper will return early, leaving the existing content intact.
+///
+/// # Example
+///
+/// The AWS config value is generated via
+/// `{{ aws_config settings.aws.config settings.aws.profile }}`
+///
+/// This would result in something like:
+/// ```toml
+/// [default]
+/// use_fips_endpoint=false
+/// ```
+///
+/// The helper will then base64 encode this content and set as `settings.aws.config`.
+pub fn aws_config(
+    helper: &Helper<'_, '_>,
+    _: &Handlebars,
+    _: &Context,
+    renderctx: &mut RenderContext<'_, '_>,
+    out: &mut dyn Output,
+) -> Result<(), RenderError> {
+    trace!("Starting aws-config-fips-endpoint helper");
+    let template_name = template_name(renderctx);
+
+    check_param_count(helper, template_name, 2)?;
+
+    let aws_config = get_param(helper, 0)?;
+    if let Value::Null = aws_config {
+        // get the profile parameter, which is probably given by the template value
+        // settings.aws.profile. Regardless, it is expected to be a string.
+        let aws_profile = get_param(helper, 1)?;
+        let aws_profile = aws_profile
+            .as_str()
+            .with_context(|| error::AwsProfileSnafu {
+                value: aws_profile.to_owned(),
+                template: template_name,
+            })?;
+        let fips_enabled = fips_enabled();
+
+        // construct the base64 encoded AWS config
+        let new_aws_config = build_aws_config(aws_profile, fips_enabled);
+
+        // write it to the template
+        out.write(&new_aws_config)
+            .with_context(|_| error::TemplateWriteSnafu {
+                template: template_name.to_owned(),
+            })?;
+    }
+
+    Ok(())
+}
+
+/// Constructs the base64 encoded AWS config for a variant, setting
+/// `use_fips_endpoint`.
+fn build_aws_config<S: AsRef<str>>(profile: S, fips_enabled: bool) -> String {
+    let aws_config_str = format!(
+        r#"[{}]
```

Comment:
Assuming that the AWS SDKs for Rust and Go expect [the same syntax as the AWS CLI](https://docs.aws.amazon.com/cli/v1/userguide/cli-configure-files.html), we have to render this differently for default and non-default profiles:
```
[default]
use_fips_endpoint = true

[profile other]
use-fips-endpoint = true
```

I'm not sure if `[profile default]` is proper syntax, though it's mentioned in [the Bottlerocket docs](https://bottlerocket.dev/en/os/1.25.x/api/settings/aws/#credentials). I would guess it means the profile that's used if you specify `--profile default` rather than the configuration that's used if you do not specify a profile. I would avoid writing a config file that uses this syntax since it's confusing even if it's not incorrect.

However, I also don't know how functional a named profile could be without a corresponding config file being supplied also. So we might want to do this instead:
* if profile is null, assume "default"
* if profile is "default", and config is null, render this config
* if profile is anything else, skip it - assume the admin knows what they're doing

Reply:
> So we might want to do this instead:

> if profile is null, assume "default"
if profile is "default", and config is null, render this config
if profile is anything else, skip it - assume the admin knows what they're doing

This sounds reasonable to me; the current implementation accounts for a variant with `settings.aws.profile` set to something non-default, while `settings.aws.config` is unset. But it wouldn't make much sense to specify a non-default profile without a corresponding config/credential file.

Reply:
I'm not seeing the kebab case `use-fips-endpoint = true` in the docs you linked. My read leads me to believe the setting is `use_fips_endpoint` regardless of profile

Reply:
> I'm not seeing the kebab case `use-fips-endpoint = true` in the docs you linked. My read leads me to believe the setting is `use_fips_endpoint` regardless of profile

Oh sorry, that was an error in my example; I'm used to kebab case for Bottlerocket. All I meant to call attention to was the use of `[default]` for default settings, and `[profile x]` for profile-specific settings, and not `[profile default]` for anything.